overtime/algorithms/edgeDeletion.py

```diff
@@ -0,0 +1,285 @@
+                # update the destination node's data.
+                destination.data['foremost_time'] = edge.end
+
+                # update accumulator
+                accumulator += 1
+                if accumulator == h:
+                    break
+
+        elif edge.start >= end:
+            # stop the algorithm.
+            break
+
+    # return the resulting foremost tree.
+    return tree
+
+
+def h_approximation(graph, h):
+    """
+        A h-approximation algorithm to return a set of edges E_ such that (G,λ)\E_ has temporal reachability at most h.
+
+        Parameter(s):
+        -------------
+        graph : TemporalDiGraph
+                A directed, temporal graph.
+        h : int
+            the maximum permitted temporal reachability.
+
+        Returns:
+        --------
+        E_ : list
+            a set of edges such that (G,λ)/E_ has temporal reachability at most h.
+
+    """
+
+    E_ = []
+
+    # find the node whose temporal reachability is more than h
+    while (True):
+        root = ''
+        for node in graph.nodes.set:
+            reachability = ot.calculate_reachability(graph, node.label)
+            if reachability > h:
+                root = node.label
+                break
+
+        # check if the specified root actually exists in the graph.
+        if root == '':
+            print('there is no node whose reachability is more than {}'.format(h))
+            break
+
+        # generate a reachable subtree based on the root
+        subtree = reachable_subtree(graph, root, h)
+
+        E_.extend(subtree.edges.uids())
+        # update (G, λ) ← (G, λ) \ E_
+        for edge in subtree.edges.uids():
+            graph.remove_edge(edge)
+
+    return E_
+
+def max_reachability(graph):
+    """
+        A method to calculate the max temporal reachability of a network.
+
+        Parameter(s):
+        -------------
+        graph : TemporalDiGraph
+                A directed, temporal graph.
+
+        Returns:
+        --------
+        maxReachability : int
+            the max temporal reachability of a network.
+
+    """
+
+    maxReachability = 0
+
+    # check the temporal reachability of each node
+    for node in graph.nodes.labels():
+        r = ot.calculate_reachability(graph, node)
+        if r > maxReachability:
+            maxReachability = r
+
+    return maxReachability
+
+
+def max_endtime(graph):
+    """
+        A method to calculate the maximum endtime of a network.
+
+        Parameter(s):
+        -------------
+        graph : TemporalDiGraph
+                A directed, temporal graph.
+
+        Returns:
+        --------
+        maxEndtime : int
+            the maximum endtime of a network.
+
+    """
+
+    maxEndtime = 0
+
+    # check the end time of each edges
+    for edge in graph.edges.set:
+        if edge.end > maxEndtime:
+            maxEndtime = edge.end
+
+    return maxEndtime
+
+
+def find_edges(graph, layout, j):

```

Comment:
The function is not clear without the context.

0) Describe what function takes as an input, what it does, and what it outputs.
1) Please provide detailed comments what is layout, what is j, etc.
2) Probably rename the function so that the name of the function is more self-explaining

Reply:
The layout is a linear representation of the nodes of graph, such as v1,...,vn. And j is an indicator. This function will return all edges which span v1,...,vj and vj+1,...,vn. I'm not sure what's the suitable name of this function.